snuba/web/split.py

```diff
@@ -13,39 +19,65 @@
+    query_limit = query.get_limit() or 0
+    limit = query_limit if query_limit is not None else 0
+
+    # cannot return (use_split and limit) since mypy expects us to return
+    # a boolean. use_split would be Union[Any, Int]
+    return (use_split or 0) > 0 and limit > 0 and not query.get_groupby()
+
+
+def _identify_condition(condition: Any, field: str, operand: str) -> bool:
+    return is_condition(condition) and condition[0] == field and condition[1] == operand
+
+
+def _replace_condition(
+    query: Query, field: str, operand: str, new_literal: Union[str, List[Any]]
+) -> None:
+    query.set_conditions(
+        [
+            cond
+            if not _identify_condition(cond, field, operand)
+            else [field, operand, new_literal]
+            for cond in query.get_conditions() or []
+        ]
+    )
+
+
+class TimeSplitQueryStrategy(StorageQuerySplitStrategy):
+    """
+    A strategy that breaks the time window into smaller ones and executes
+    them in sequence.
+    """
+
+    def __init__(self, timestamp_col: str) -> None:
+        self.__timestamp_col = timestamp_col
+
+    def can_execute(self, request: Request) -> bool:
         remaining_offset = request.query.get_offset()
         orderby = util.to_list(request.query.get_orderby())
 
-        common_conditions = use_split and limit and not request.query.get_groupby()
-
-        if common_conditions:
-            total_col_count = len(request.query.get_all_referenced_columns())
-            column_split_spec = dataset.get_split_query_spec()
-            if column_split_spec:
-                copied_query = copy.deepcopy(request.query)
-                copied_query.set_selected_columns(column_split_spec.get_min_columns())
-                min_col_count = len(copied_query.get_all_referenced_columns())
-            else:
-                min_col_count = None
+        has_from_date = any(
+            _identify_condition(condition, self.__timestamp_col, ">=")
+            for condition in request.query.get_conditions() or []
+        )
 
-            if (
-                column_split_spec
-                and request.query.get_selected_columns()
-                and not request.query.get_aggregations()
-                and total_col_count > min_col_count
-            ):
-                return col_split(dataset, request, column_split_spec, *args, **kwargs)
-            elif orderby[:1] == ["-timestamp"] and remaining_offset < 1000:
-                return time_split(dataset, request, *args, **kwargs)
+        has_to_date = any(
+            _identify_condition(condition, self.__timestamp_col, "<")
+            for condition in request.query.get_conditions() or []
+        )
 
-        return query_func(dataset, request, *args, **kwargs)
+        return (
+            _is_query_splittable(request.query)
+            and has_from_date
+            and has_to_date
+            and orderby[:1] == [f"-{self.__timestamp_col}"]
```

Comment:
right, I will do a cleanup pass on the split algorithm code. 
Initially I wanted to touch the existing code as little as possible to avoid making reviewers to  cope with changes to the splitting algorithm. But by removing the dependency on the extensions it seems the algorithm was changed anyway so it may look confusing now.